core/indigo-core/molecule/meta_commons.h

```diff
@@ -38,29 +39,170 @@ namespace indigo
 {
     const double KDefaultFontSize = 13;
     const double KFontScaleFactor = 47;
-    const auto KFontBoldStr = "BOLD";
-    const auto KFontItalicStr = "ITALIC";
-    const auto KFontSuperscriptStr = "SUPERSCRIPT";
-    const auto KFontSubscriptStr = "SUBSCRIPT";
-    const auto KFontCustomSizeStr = "CUSTOM_FONT_SIZE";
+    const auto KFontBoldStrV1 = "BOLD";
+    const auto KFontItalicStrV1 = "ITALIC";
+    const auto KFontSuperscriptStrV1 = "SUPERSCRIPT";
+    const auto KFontSubscriptStrV1 = "SUBSCRIPT";
+    const auto KFontCustomSizeStrV1 = "CUSTOM_FONT_SIZE";
     const auto KImagePNG = "image/png";
     const auto KImageSVG = "image/svg+xml";
 
+    const auto KFontBoldStr = "bold";
+    const auto KFontItalicStr = "italic";
+    const auto KFontSuperscriptStr = "superscript";
+    const auto KFontSubscriptStr = "subscript";
+    const auto KFontSizeStr = "size";
+    const auto KFontColorStr = "color";
+    const auto KFontFamilyStr = "family";
+
+    const auto KAlignmentLeft = "left";
+    const auto KAlignmentRight = "right";
+    const auto KAlignmentCenter = "center";
+    const auto KAlignmentFull = "full";
+
     const uint8_t KReactantArea = 0;
     const uint8_t KReagentUpArea = 1;
     const uint8_t KReagentDownArea = 2;
     const uint8_t KProductArea = 3;
     const Vec2f MIN_MOL_SIZE = {0.5, 0.5};
 
+    struct KETVersion
+    {
+        int major;
+        int minor;
+        int patch;
+    };
+
+    const KETVersion KETVersion1 = {1, 0, 0};
+    const KETVersion KETVersion2 = {2, 0, 0};
+
+    enum class KETVersionIndex : int
+    {
+        EMajor,
+        EMinor,
+        EPatch
+    };
+
+    using KETFontVal = std::variant<std::monostate, std::string, uint32_t>;
+
+    struct KETFontStyleStatus
+    {
+        KETFontStyleStatus() : offset(0), size(0), val(std::monostate{})
+        {
+        }
+
+        KETFontStyleStatus(std::size_t off, std::size_t sz, KETFontVal v) : offset(off), size(sz), val(v)
+        {
+        }
+
+        std::size_t offset;
+        std::size_t size;
+        KETFontVal val;
+    };
+
+    struct KETFontStyle
+    {
+        enum class FontStyle : int
+        {
+            ENone,
+            EBold,
+            EItalic,
+            ESuperScript,
+            ESubScript,
+            EFamily,
+            ESize,
+            EColor
+        };
+
+        KETFontStyle(const FontStyle fs, KETFontVal val = std::monostate{}) : _font_style(fs), _val(val)
+        {
+        }
+
+        KETFontStyle() : _font_style(FontStyle::ENone)
+        {
+        }
+
+        KETFontStyle(const KETFontStyle& other) : _font_style(other._font_style), _val(other._val)
+        {
+        }
+
+        operator int() const
```

Comment:
For me better use explicit function like `intFontStyle`

Reply:
fixed